### PR TITLE
chore: add python 3.12 support

### PR DESCRIPTION
**Pull request by Ian2012**
### Description

This PR removes the dependency of `pkg_resources` for python3.12 compatibility

### CODE REVIEW

File: drydock/plugin.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 import functools
 import os
```
Comment (jfavellar90) [inline drydock/plugin.py:3]: Please add the Python [support](https://github.com/eduNEXT/drydock/blob/main/setup.py#L53) for 3.11 and 3.12 as [Tutor](https://github.com/overhangio/tutor/blob/master/setup.py#L75) does

File: drydock/plugin.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
...
 import click
-import pkg_resources
+import importlib_resources
```
Comment (MoisesGSalas) [inline drydock/plugin.py:5]: Why `importlib_resources` over `importlib.resources`?

Comment (Ian2012) [reply drydock/plugin.py:5]: This is what is used by all tutor [plugins](https://github.com/overhangio/cookiecutter-tutor-plugin/blob/master/%7B%7B%20cookiecutter.package_name%20%7D%7D/%7B%7B%20cookiecutter.module_name%20%7D%7D/plugin.py#L7).

Comment (MoisesGSalas) [reply drydock/plugin.py:5]: Seems like the `importlib_resources` package was added to tutor dependencies in [`v17.0.2`](https://github.com/overhangio/tutor/commit/d99b2feeb30fa772dca48345c8992b4eafdeac21) does this break on previous tutor versions?

Comment (Ian2012) [reply drydock/plugin.py:5]: Yes, it breaks:

image

Comment (Ian2012) [reply drydock/plugin.py:5]: However, this is not a problem as this change is specific the drydock latest version

Comment (MoisesGSalas) [reply drydock/plugin.py:5]: But we should still update the setup.py (https://github.com/eduNEXT/drydock/blob/main/setup.py#L42).

**Review by MoisesGSalas — APPROVED** [review]

**Pull request merged**